llvm/lib/Analysis/VFABIDemangling.cpp

```diff
@@ -66,24 +66,26 @@ ParseRet tryParseMask(StringRef &MangledName, bool &IsMasked) {
 /// vector length. On success, the `<vlen>` token is removed from
 /// the input string `ParseString`.
 ///
-ParseRet tryParseVLEN(StringRef &ParseString, unsigned &VF, bool &IsScalable) {
+ParseRet tryParseVLEN(StringRef &ParseString,
```

Comment:
Sorry, my suggest here is bogus because you cannot create a zero length scalable vector.  What I'm really asking for is whether it's possible to return some kind of token that is later used to contract the ElementCount once all other information is available.  Perhaps this is just a local std::pair that rather resembles an ElementCount?

Reply:
That's pretty much the same thing expressed a different way -- we either have a valid non-zero fixed length VF, or we know that the VF is scalable but the minimum is yet to be determined.

It's a pity std::variant is so limited; something like Rust's enums would be great for this (or any language with sum types).

Maybe we could return a variant that's either an unsigned for a fixed VF, a dummy bool for scalable, or a ParseRet for failure?